tests/test_retrieve.py

```diff
@@ -61,6 +61,15 @@ def mock_boto3_client():
         yield mock_client
 
 
+@pytest.fixture(autouse=True)
+def clear_aws_region_env():
```

Comment:
We should be mocking the environment instead of using the os.environment.

See https://github.com/strands-agents/tools/blob/3970a66d7f38bdce3c88e78e809db03552d0ed9b/tests/test_environment.py#L29 for an example